nginx-asg.tf

```diff
@@ -0,0 +1,194 @@
+  launch_configuration      = aws_launch_configuration.nginx_conf.name
+  vpc_zone_identifier       = data.terraform_remote_state.internal_compute.outputs.bgdc_subnet.ids
+  target_group_arns         = [aws_lb_target_group.dwx_bdgc_nginx_nlb_tg.arn]
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  tag {
+    key                 = "Name"
+    value               = "dwx-bgdc-nginx-instances"
+    propagate_at_launch = true
+  }
+
+}
+
+
+resource "aws_lb" "dwx_bdgc_nginx_nlb" {
+  name               = "dwx-bgdc-nginx-nlb"
+  internal           = true
+  load_balancer_type = "network"
+  subnets            = data.terraform_remote_state.internal_compute.outputs.bgdc_subnet.ids
+
+
+  enable_deletion_protection = false
+
+  tags = {
+    Environment = local.environment
+  }
+}
+
+resource "aws_lb_target_group" "dwx_bdgc_nginx_nlb_tg" {
+  name     = "dwx-bgdc-nginx-nlb-tg"
+  port     = 400
+  protocol = "TCP"
+  target_type = "instance"
+  vpc_id   = data.terraform_remote_state.internal_compute.outputs.vpc.vpc.vpc.id
+
+  health_check {
+    port = 80
+    protocol          = "TCP"
+    healthy_threshold = 3
+    unhealthy_threshold = 3
+    timeout = 10
+    interval = 30
+  }
+}
+
+resource "aws_lb_listener" "dwx_bdgc_nginx_nlb_listener" {
+  load_balancer_arn = aws_lb.dwx_bdgc_nginx_nlb.arn
+  port              = "400"
+  protocol          = "TCP"
+
+  default_action {
+    target_group_arn = aws_lb_target_group.dwx_bdgc_nginx_nlb_tg.arn
+    type             = "forward"
+  }
+}
+
+
+resource "aws_autoscaling_attachment" "asg_attachment_nginx_nlb" {
+  autoscaling_group_name = aws_autoscaling_group.nginx_asg.id
+  alb_target_group_arn   = aws_lb_target_group.dwx_bdgc_nginx_nlb_tg.id
+}
+
+
+resource "aws_vpc_endpoint_service" "bgdc_dwx_end_point_service" {
+  acceptance_required        = false
+  network_load_balancer_arns = [aws_lb.dwx_bdgc_nginx_nlb.arn]
+  allowed_principals = ["arn:aws:iam::${local.bgdc_account.test}:root"]
+  tags = {
+    Name = "bgdc-dwx-endpoint-svc"
+  }
+}
+
+output "bgdc_dwx_vpc_endpoint_service_name" {
+  value = aws_vpc_endpoint_service.bgdc_dwx_end_point_service.service_name
+}
+
+
+resource "aws_security_group" "nginx-bgdc-dwx" {
+  name                   = "nginx-bgdc-dwx-instances"
+  description            = "Contains rules for nginx instances"
+  revoke_rules_on_delete = true
+  vpc_id                 = data.terraform_remote_state.internal_compute.outputs.vpc.vpc.vpc.id
+}
+
+
+resource "aws_security_group_rule" "allow_http_from_target_group" {
+  description              = "HTTP from target group"
+  from_port                = 80
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.nginx-bgdc-dwx.id
+  to_port                  = 80
+  type                     = "ingress"
+  cidr_blocks              = formatlist("%s/32", [for eni in data.aws_network_interface.lb : eni.private_ip])
+    
+                             
+}
+
+
+data "aws_network_interface" "lb" {
```

Comment:
Something with a more meaningful name that "lb". Again, in case people refer to remote states, this won't be clear.